ttnn/cpp/ttnn/operations/data_movement/common/common.cpp

```diff
@@ -19,11 +21,53 @@ ttnn::Tensor squeeze_to_le_4D(const ttnn::Tensor& tensor) {
         while (rank > 4) {
             squeezed = ttnn::squeeze(squeezed, 0);
             rank = squeezed.get_shape().rank();
+            printf("rank: %zu\n", rank);
         }
         return squeezed;
     }
 };
 
+ttnn::Shape squeeze_output_shape(ttnn::Shape output_shape) {
+    if (output_shape.rank() > 4) {
+        std::array<uint32_t, 4> output_shape_4d;
+        output_shape_4d[0] = 1;
+        int extra_rank = output_shape.rank() - 4;
+        for (int i = extra_rank; i >= 0; i--) {
+            output_shape_4d[0] *= output_shape[i];
+        }
+        output_shape_4d[1] = output_shape[1 + extra_rank];
+        output_shape_4d[2] = output_shape[2 + extra_rank];
+        output_shape_4d[3] = output_shape[3 + extra_rank];
+        return ttnn::Shape(output_shape_4d);
+    }
+    return output_shape;
+}
+
+ttnn::Tensor squeeze_from_ND_to_4D(const ttnn::Tensor& tensor) {
+    auto shape = tensor.get_shape();
+    auto rank = shape.rank();
+    if (shape.rank() < 4) {
+        TT_THROW("Tensor has to be of rank larger than 4!");
+    }
+    if (rank == 4) {
+        return tensor;
+    }
+    int i = 0;
+    if (shape[i] == 1) {
```

Comment:
How does this work for at least 6D where the first dimension is 1 and the second one is not?

E.g. `[1, 3, 64, 1, 32, 1280]` from one of the tests or `[1, 32, 32, 32, 64, 1, 32, 1280]`?

And if that works without having to go through the while loop, why would it be necessary to through the while loop for e.g. `[32, 64, 1, 32, 1280]`?

Reply:
For tensors with dim>=6D, as long as the shape value is 1 at first, the tensor is squeezed, then it is reshaped when the shape value changes. For [1,3,64,1,32,1280] for example, the tensor is squeezed once, then we call ttnn::reshape. Theoretically, we don't need to check if shape[i] == 1, and the behaviour should be the same with just reshape operations for these tests. However, for a different post commit test that calls tilize explicitly for an ND tensor starting with shape 1, squeeze is required for the test to work.

Reply:
Could you please elaborate on this other test?

Why does it e.g. require squeeze for `[1,3,64,1,32,1280]` and not for `[2,3,64,1,32,1280]`?